repose-aggregator/components/services/http-client-service/http-client-service-impl/src/main/java/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolProvider.java

```diff
@@ -83,6 +88,39 @@ public static HttpClient genClient(PoolType poolConf) {
         DefaultHttpClient client = new DefaultHttpClient(cm, params);
 
         SSLContext sslContext = ProxyUtilities.getTrustingSslContext();
+
+        if (poolConf.getKeystoreFilename() != null) {
+            SSLContextBuilder sslContextBuilder = SSLContexts.custom();
+            try {
+                File keystoreFile = new File(poolConf.getKeystoreFilename());
+                if (!keystoreFile.isAbsolute()) {
+                    keystoreFile = new File(configRoot, poolConf.getKeystoreFilename());
```

Comment:
you don't need to do this if, this line will always behave correctly, or at least that's what the docs say.